src/Drivers/check_spo.cpp

```diff
@@ -162,8 +162,8 @@ int main(int argc, char **argv)
     tileSize              = (tileSize > 0) ? tileSize : norb;
     nTiles                = norb / tileSize;
 
-    const unsigned int SPO_coeff_size =
-        (nx + 3) * (ny + 3) * (nz + 3) * norb * sizeof(RealType);
+    const size_t SPO_coeff_size =
+        (size_t)norb * (nx + 3) * (ny + 3) * (nz + 3) * sizeof(RealType);
```

Comment:
The size_t should be unecessary, do you have a compiler that requires it? The fewer unecessary casts the better.  Same for the second one.

Reply:
Do you mean the ```(size_t)norb```? I was not sure that all compilers will do 64bit integer.

Reply:
Yes that is the one I mean my understanding of C++'s implicit conversion is all the int should be getting promoted to size_t if its the larger type.

Reply:
Could you point me the standard? I'd like to have a read.

Reply:
I checked without this cast, I got 6388.59MB. Still wrong. This cast is needed.
For my understanding of the implicit conversion, only the final assign promote the type without the explicit casting.

Reply:
Here's the cppreference
https://en.cppreference.com/w/cpp/language/operator_arithmetic#Conversions
I'm not sure which standards docs are key.

Reply:
It seems consistent with what I saw. The left to right rule applies here.

Reply:
Ah I see, I think lisp has made me a little worse at C++

Reply:
Prefer static_cast<size_t>() to (size_t) for compile time check?

Reply:
Intel on my workstation gives all these numbers. Only when both changes applied I get the correct number.

Reply:
You are correct sir left to right applies so a cast on the first value is necessay.

```
#include <limits>
#include <iostream>

int main()
{
  size_t sizeof_size = std::numeric_limits<int>::max() * std::numeric_limits<int>::max() * sizeof(int);
  std::cout << sizeof_size << '\n';
  unsigned int tu_size = (unsigned int)std::numeric_limits<int>::max() * std::numeric_limits<int>::max();
  std::cout << tu_size << '\n';
  size_t test_size = static_cast<size_t>(std::numeric_limits<int>::max()) * std::numeric_limits<int>::max();
  std::cout << test_size << '\n';
}
```

```
bash-4.4$ g++-7 test.cpp
test.cpp: In function 'int main()':
test.cpp:6:56: warning: integer overflow in expression [-Woverflow]
   size_t sizeof_size = std::numeric_limits<int>::max() * std::numeric_limits<int>::max() * sizeof(int);
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
bash-4.4$ ./a.out
4
1
4611686014132420609
bash-4.4$ 
```